virttest/utils_net.py

```diff
@@ -3744,3 +3746,190 @@ def map_hostname_ipaddress(hostname_ip_dict, session=None):
                 return False
     logging.info("All the hostnames and IPs are mapped in %s", hosts_file)
     return True
+
+
+def _remove_file(session, file_path):
+    """
+    Remove a file at file_path
+
+    :param file_path: file path to remove
+    :param session: a logined session
+    """
+
+    clean_cmd = "del %s" % file_path
+    session.cmd_status(clean_cmd)
+
+
+def _copy_file(session, src_path, dst_folder="c:\\"):
+    """
+    Copy a file at src_path to dst_folder.
+    If a same name file already exists at dst_folder, it will be replaced.
+
+    :param session: a logined sesison
+    :param src_path: file to copy from
+    :param dst_folder: folder to copy to
+
+    """
+
+    copy_cmd = "xcopy %s %s /y" % (src_path, dst_folder)
+    session.cmd_status(copy_cmd)
+
+
+def _get_traceview_path(session, params):
+    """
+    :param session: a session to send cmd
+    :param params: the test params
+    :return: the proper traceview path
+    """
+
+    traceview_path_template = "WIN_UTILS:\\traceview\\%s\\%s\\traceview.exe"
+    # traceview_ver = params["traceview_version"]
+    traceview_ver = "win10"
+    os_version = system.version(session)
+    main_ver = int(os_version.split('.')[0])
+    if main_ver < 10:
+        traceview_ver = "win8"
+    os_arch = "%PROCESSOR_ARCHITECTURE%"
+    traceview_path_template = traceview_path_template % (traceview_ver, os_arch)
+    return utils_misc.set_winutils_letter(session, traceview_path_template)
+
+
+def _prepare_traceview_windows(test, params, vm, session, timeout=360):
+    """
+    Copy traceview.exe and corresponding pdb file to drive c: for future use.
+
+    :param test: the running test
+    :param params: the test params
+    :param vm: target vm
+    :param session: a session to send command
+    :param timeout: the command execute timeout
+    """
+
+    # copy traceview.exe
+    logging.info("Copy traceview.exe to drive C")
+    traceview_path = _get_traceview_path(session, params)
+    _copy_file(session, traceview_path)
+
+    # copy Netkvm.pdb
+    logging.info("Find Netkvm.pdb and copy to drive C")
+    viowin_ltr = virtio_win.drive_letter_iso(session)
+    if not viowin_ltr:
+        logging.error("Could not find virtio-win drive in guest")
+    guest_name = virtio_win.product_dirname_iso(session)
+    if not guest_name:
+        logging.error("Could not get product dirname of the vm")
+    guest_arch = virtio_win.arch_dirname_iso(session)
+    if not guest_arch:
+        logging.error("Could not get architecture dirname of the vm")
+
+    driver_name = params.get("driver_name", "netkvm")
+    pdb_middle_path = "%s\\%s" % (guest_name, guest_arch)
+    pdb_find_cmd = 'dir /b /s %s\\%s.pdb | findstr "\\%s\\\\"'
+    pdb_find_cmd %= (viowin_ltr, driver_name, pdb_middle_path)
+    pdb_path = session.cmd(pdb_find_cmd, timeout=timeout).strip()
+    logging.info("Found Netkvm.pdb file at '%s'", pdb_path)
+    _copy_file(session, pdb_path)
```

Comment:
Better return the full path of traceview and pdb in this function, instead of hard coding at L#3899-3900.